packages/plugin-fontawesome/src/FontAwesomeHtmlDomRenderer.ts

```diff
@@ -12,10 +12,7 @@ export class FontAwesomeHtmlDomRenderer extends AbstractRenderer<Node[]> {
 
     async render(node: FontAwesomeNode): Promise<Node[]> {
         const fontawesome = document.createElement(node.htmlTag);
-        const attributes = node.modifiers.find(Attributes);
-        if (attributes) {
-            this.engine.renderAttributes(attributes, fontawesome);
-        }
+        this.engine.renderAttributes(node.modifiers.get(Attributes), fontawesome);
```

Comment:
A red light appear in my mind as I realize that simply *rendering* a node is going to *modify* it with this code. I suggest reverting the changes to the renderers. The `renderAttributes` 4-liners will disappear in the future when the modifiers are rendered separately rather than being the responsibility of the node renderers. This comment obviously applies to all renderers.

EDIT: I just thought about something. What if `renderAttributes` received the `node` instead ? Then it could do the `modifiers.find` by itself and still be a one-liner in the renderers. It would be more correct to just revert to the 4-liners, but it pains me to have them all over the place when I know they are not supposed to be there in my world where the modifiers are already rendered indenpendenty. What do you think ? Plague or cholera ?

Reply:
> What if renderAttributes received the node instead ?

As great as that sounds I don't thing it's going to be possible. What about the cases where it's not `Attributes` we want to render but some extension of `Attributes`.

Eg., in `ListItemHtmlDomRenderer`:
```ts
this.engine.renderAttributes(node.modifiers.get(ListItemAttributes), domListItem);
```

Reply:
find

Reply:
?